mcp/src/main/java/io/modelcontextprotocol/client/transport/HttpClientSseClientTransport.java

```diff
@@ -392,43 +385,43 @@ public void onError(Throwable error) {
 	 */
 	@Override
 	public Mono<Void> sendMessage(JSONRPCMessage message) {
-		if (isClosing) {
+		if (state.get() == TransportState.CLOSING || state.get() == TransportState.DISCONNECTED) {
 			return Mono.empty();
 		}
-
-		try {
-			if (!closeLatch.await(10, TimeUnit.SECONDS)) {
-				return Mono.error(new McpError("Failed to wait for the message endpoint"));
+		return Mono.defer(() -> {
+			if (messageEndpoint.get() == null) {
+				return Mono.error(new McpError("No message endpoint available"));
 			}
-		}
-		catch (InterruptedException e) {
-			return Mono.error(new McpError("Failed to wait for the message endpoint"));
-		}
 
-		String endpoint = messageEndpoint.get();
-		if (endpoint == null) {
-			return Mono.error(new McpError("No message endpoint available"));
-		}
+			return serializeMessage(message).flatMap(body -> sendHttpPost(messageEndpoint.get(), body))
+				.doOnNext(this::logIfNotOk)
+				.doOnError(err -> logger.error("Error sending message", err))
+				.then();
+
+		}).retryWhen(Retry.fixedDelay(3, Duration.ofSeconds(3)).filter(err -> messageEndpoint.get() == null));
```

Comment:
See another PR with this same pattern. 3/3 retry. Maybe place this in a constant place.